spec/controllers/application_controller_spec.rb

```diff
@@ -122,4 +122,12 @@ def index
       end
     end
   end
+
+  describe '#analytics' do
+    it 'uses the Analytics class by default' do
+      expect(Analytics).to receive(:new).with(controller.current_user, request)
+
+      controller.analytics
```

Comment:
Could you re-write this as an expectation that the return value's class is `Analytics`?